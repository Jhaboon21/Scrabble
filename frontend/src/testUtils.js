import React from "react";
import UserContext from "./routes/auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "John",
  last_name: "Doe",
  email: "test@email.com",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };