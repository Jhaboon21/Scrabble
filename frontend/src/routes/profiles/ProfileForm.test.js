import React from "react";
import { render } from "@testing-library/react";
import Profile from "./ProfileForm";
import { UserProvider } from "../../testUtils";

const currentUser = {
  username: 'testuser',
  firstName: 'John',
  lastName: 'Doe',
  email: 'test@email.com',
};

it("matches snapshot", function () {
  const { asFragment } = render(
    <UserProvider currentUser={currentUser}>
      <Profile />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders profile form correctly", function () {
  const { getByLabelText, getByText } = render(
    <UserProvider currentUser={currentUser}>
      <Profile />
    </UserProvider>
  );

  expect(getByLabelText('First Name')).toHaveValue('John');
  expect(getByLabelText('Last Name')).toHaveValue('Doe');
  expect(getByLabelText('Email')).toHaveValue('test@email.com');
  expect(getByLabelText('Confirm password to make changes:')).toBeInTheDocument();
  expect(getByText('Save Changes')).toBeInTheDocument();
});
