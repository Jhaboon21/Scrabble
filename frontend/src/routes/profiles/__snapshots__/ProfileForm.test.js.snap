// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-md-6 col-lg-4 offset-md-3 offset-lg-4"
  >
    <h3>
      Profile
    </h3>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form>
          <div
            class="form-group"
          >
            <label>
              Username
            </label>
            <p
              class="form-control-plaintext"
            >
              testuser
            </p>
          </div>
          <div
            class="form-group"
          >
            <label
              for="firstName"
            >
              First Name
            </label>
            <input
              class="form-control"
              id="firstName"
              name="firstName"
              value="John"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="lastName"
            >
              Last Name
            </label>
            <input
              class="form-control"
              id="lastName"
              name="lastName"
              value="Doe"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              value="test@email.com"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="password"
            >
              Confirm password to make changes:
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-primary btn-block mt-4"
          >
            Save Changes
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
